src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs

```diff
@@ -33,15 +34,40 @@ public class OpenConnectionMessage : ConnectionMessage
         /// </summary>
         /// <param name="connectionId">The connection Id.</param>
         /// <param name="claims">An array of <see cref="Claim"/> associated with the connection.</param>
-        public OpenConnectionMessage(string connectionId, Claim[] claims) : base(connectionId)
+        public OpenConnectionMessage(string connectionId, Claim[] claims)
+            : this(connectionId, claims, new HeaderDictionary(), string.Empty)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OpenConnectionMessage"/> class.
+        /// </summary>
+        /// <param name="connectionId">The connection Id.</param>
+        /// <param name="claims">An array of <see cref="Claim"/> associated with the connection.</param>
+        /// <param name="headers">A <see cref="IHeaderDictionary"/> associated with the connection.</param>
+        /// <param name="queryString">Query string associated with the connection.</param>
+        public OpenConnectionMessage(string connectionId, Claim[] claims, IHeaderDictionary headers, string queryString)
+            : base(connectionId)
         {
             Claims = claims;
+            Headers = headers;
+            QueryString = queryString;
         }
 
         /// <summary>
         /// Gets or sets the associated claims.
         /// </summary>
         public Claim[] Claims { get; set; }
+
+        /// <summary>
+        /// Gets or sets the associated headers.
+        /// </summary>
+        public IHeaderDictionary Headers { get; set; }
```

Comment:
OK giving it some more thought, I think this should be `IDictionary<string, StringValues>` and we should change this dependency `Microsoft.AspNetCore.Http` to `Microsoft.Extensions.Primitives`

Reply:
Any reason other than minimum dependency?

Reply:
Nope It’s about minimizing dependencies